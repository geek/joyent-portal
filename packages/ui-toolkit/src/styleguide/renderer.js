import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
