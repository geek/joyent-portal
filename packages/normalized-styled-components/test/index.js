import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {
  Article,
  Aside,
  Footer,
  Header,
  Nav,
  Section,
  FigCaption,
  Figure,
  Main,
  H1,
  Hr,
  Pre,
  A,
  Abbr,
  B,
  Strong,
  Code,
  Kbd,
  Samp,
  Dfn,
  Mark,
  Small,
  Sub,
  Sup,
  Audio,
  Video,
  Img,
  Svg,
  Button,
  Input,
  Optgroup,
  Select,
  Textarea,
  Fieldset,
  Legend,
  Progress,
  Details,
  Menu,
  Summary,
  Canvas
} from '../src/';

it('renders <Article /> correctly', () => {
  const tree = renderer.create(<Article />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Aside /> correctly', () => {
  const tree = renderer.create(<Aside />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Footer /> correctly', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Header /> correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Nav /> correctly', () => {
  const tree = renderer.create(<Nav />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Section /> correctly', () => {
  const tree = renderer.create(<Section />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <FigCaption /> correctly', () => {
  const tree = renderer.create(<FigCaption />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Figure /> correctly', () => {
  const tree = renderer.create(<Figure />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Main /> correctly', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <H1 /> correctly', () => {
  const tree = renderer.create(<H1 />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Hr /> correctly', () => {
  const tree = renderer.create(<Hr />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Pre /> correctly', () => {
  const tree = renderer.create(<Pre />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <A /> correctly', () => {
  const tree = renderer.create(<A />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Abbr /> correctly', () => {
  const tree = renderer.create(<Abbr />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <B /> correctly', () => {
  const tree = renderer.create(<B />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Strong /> correctly', () => {
  const tree = renderer.create(<Strong />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Code /> correctly', () => {
  const tree = renderer.create(<Code />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Kbd /> correctly', () => {
  const tree = renderer.create(<Kbd />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Samp /> correctly', () => {
  const tree = renderer.create(<Samp />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Dfn /> correctly', () => {
  const tree = renderer.create(<Dfn />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Mark /> correctly', () => {
  const tree = renderer.create(<Mark />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Small /> correctly', () => {
  const tree = renderer.create(<Small />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Sub /> correctly', () => {
  const tree = renderer.create(<Sub />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Sup /> correctly', () => {
  const tree = renderer.create(<Sup />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Audio /> correctly', () => {
  const tree = renderer.create(<Audio />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Video /> correctly', () => {
  const tree = renderer.create(<Video />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Img /> correctly', () => {
  const tree = renderer.create(<Img />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Svg /> correctly', () => {
  const tree = renderer.create(<Svg />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Button /> correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Input /> correctly', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Optgroup /> correctly', () => {
  const tree = renderer.create(<Optgroup />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Select /> correctly', () => {
  const tree = renderer.create(<Select />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Textarea /> correctly', () => {
  const tree = renderer.create(<Textarea />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Fieldset /> correctly', () => {
  const tree = renderer.create(<Fieldset />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Legend /> correctly', () => {
  const tree = renderer.create(<Legend />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Progress /> correctly', () => {
  const tree = renderer.create(<Progress />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Details /> correctly', () => {
  const tree = renderer.create(<Details />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Menu /> correctly', () => {
  const tree = renderer.create(<Menu />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Summary /> correctly', () => {
  const tree = renderer.create(<Summary />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders <Canvas /> correctly', () => {
  const tree = renderer.create(<Canvas />).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});
