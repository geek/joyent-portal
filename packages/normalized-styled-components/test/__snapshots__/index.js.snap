// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <A /> correctly 1`] = `
.jleQxG {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
}

<a
  className="sc-gqjmRU jleQxG"
/>
`;

exports[`renders <Abbr /> correctly 1`] = `


<abbr
  className="sc-VigVT ijOVMO"
/>
`;

exports[`renders <Article /> correctly 1`] = `
.fpcbkF {
  display: block;
}

<article
  className="sc-bdVaJa fpcbkF"
/>
`;

exports[`renders <Aside /> correctly 1`] = `
.gbieAY {
  display: block;
}

<aside
  className="sc-bwzfXH gbieAY"
/>
`;

exports[`renders <Audio /> correctly 1`] = `
.SYMpT {
  display: inline-block;
}

.SYMpT:not([controls]) {
  display: none;
  height: 0;
}

<audio
  className="sc-ckVGcZ SYMpT"
/>
`;

exports[`renders <B /> correctly 1`] = `
.vhpOG {
  font-weight: bolder;
}

<b
  className="sc-jTzLTM vhpOG"
/>
`;

exports[`renders <Button /> correctly 1`] = `
.efaqoE {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  -webkit-text-transform: none;
  text-transform: none;
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
}

.efaqoE::-moz-focus-inner,
.efaqoE[type="button"]::-moz-focus-inner,
.efaqoE[type="reset"]::-moz-focus-inner,
.efaqoE[type="submit"]::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.efaqoE:-moz-focusring,
.efaqoE[type="button"]:-moz-focusring,
.efaqoE[type="reset"]:-moz-focusring,
.efaqoE[type="submit"]:-moz-focusring {
  outline: 0.0625rem dotted ButtonText;
}

<button
  className="sc-kEYyzF efaqoE"
/>
`;

exports[`renders <Canvas /> correctly 1`] = `
.gIGqrW {
  display: inline-block;
}

<canvas
  className="sc-gPEVay gIGqrW"
/>
`;

exports[`renders <Code /> correctly 1`] = `
.NmzmU {
  font-family: monospace, monospace;
  font-size: 1rem;
}

<code
  className="sc-jzJRlG NmzmU"
/>
`;

exports[`renders <Details /> correctly 1`] = `
.kSeyxS {
  display: block;
}

<details
  className="sc-cMljjf kSeyxS"
/>
`;

exports[`renders <Dfn /> correctly 1`] = `
.lcACZG {
  font-style: italic;
}

<dfn
  className="sc-chPdSV lcACZG"
/>
`;

exports[`renders <Fieldset /> correctly 1`] = `
.dLqrpE {
  padding: 0.35em 0.75em 0.625em;
}

<fieldset
  className="sc-cvbbAY dLqrpE"
/>
`;

exports[`renders <FigCaption /> correctly 1`] = `
.fGYiaS {
  display: block;
}

<figcaption
  className="sc-bZQynM fGYiaS"
/>
`;

exports[`renders <Figure /> correctly 1`] = `
.fpLcXm {
  display: block;
  margin: 1rem 2.5rem;
}

<figure
  className="sc-gzVnrw fpLcXm"
/>
`;

exports[`renders <Footer /> correctly 1`] = `
.dMAkQL {
  display: block;
}

<footer
  className="sc-htpNat dMAkQL"
/>
`;

exports[`renders <H1 /> correctly 1`] = `
.itlxbG {
  font-size: 2rem;
  margin: 0.625rem 0;
}

<h1
  className="sc-dnqmqq itlxbG"
/>
`;

exports[`renders <Header /> correctly 1`] = `
.imCjfa {
  display: block;
}

<header
  className="sc-bxivhb imCjfa"
/>
`;

exports[`renders <Hr /> correctly 1`] = `
.jvTISp {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}

<hr
  className="sc-iwsKbI jvTISp"
/>
`;

exports[`renders <Img /> correctly 1`] = `
.gLFqHa {
  border-style: none;
}

<img
  className="sc-eNQAEJ gLFqHa"
/>
`;

exports[`renders <Input /> correctly 1`] = `
.gtmtGl {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.gtmtGl::-webkit-file-upload-button {
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  font: inherit;
}

<input
  className="sc-kkGfuU gtmtGl"
/>
`;

exports[`renders <Kbd /> correctly 1`] = `
.cNQErp {
  font-family: monospace, monospace;
  font-size: 1rem;
}

<kbd
  className="sc-cSHVUG cNQErp"
/>
`;

exports[`renders <Legend /> correctly 1`] = `
.iBvQlR {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}

<legend
  className="sc-jWBwVP iBvQlR"
/>
`;

exports[`renders <Main /> correctly 1`] = `
.kGbKV {
  display: block;
}

<main
  className="sc-htoDjs kGbKV"
/>
`;

exports[`renders <Mark /> correctly 1`] = `
.gUcHXP {
  background-color: #ff0;
  color: #000;
}

<mark
  className="sc-kgoBCf gUcHXP"
/>
`;

exports[`renders <Menu /> correctly 1`] = `
.EtDOy {
  display: block;
}

<menu
  className="sc-jAaTju EtDOy"
/>
`;

exports[`renders <Nav /> correctly 1`] = `
.kXFzZl {
  display: block;
}

<nav
  className="sc-ifAKCX kXFzZl"
/>
`;

exports[`renders <Optgroup /> correctly 1`] = `
.bwvGAC {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}

<optgroup
  className="sc-iAyFgw bwvGAC"
/>
`;

exports[`renders <Pre /> correctly 1`] = `
.ePyiae {
  font-family: monospace, monospace;
  font-size: 1rem;
}

<pre
  className="sc-gZMcBi ePyiae"
/>
`;

exports[`renders <Progress /> correctly 1`] = `
.irOoVX {
  display: inline-block;
  vertical-align: baseline;
}

<progress
  className="sc-brqgnP irOoVX"
/>
`;

exports[`renders <Samp /> correctly 1`] = `
.brPjlW {
  font-family: monospace, monospace;
  font-size: 1rem;
}

<samp
  className="sc-kAzzGY brPjlW"
/>
`;

exports[`renders <Section /> correctly 1`] = `
.kyfnqM {
  display: block;
}

<section
  className="sc-EHOje kyfnqM"
/>
`;

exports[`renders <Select /> correctly 1`] = `
.kskmSD {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  -webkit-text-transform: none;
  text-transform: none;
}

<select
  className="sc-hSdWYo kskmSD"
/>
`;

exports[`renders <Small /> correctly 1`] = `
.eNDzVi {
  font-size: 80%;
}

<small
  className="sc-kGXeez eNDzVi"
/>
`;

exports[`renders <Strong /> correctly 1`] = `
.liZvtt {
  font-weight: bolder;
}

<strong
  className="sc-fjdhpX liZvtt"
/>
`;

exports[`renders <Sub /> correctly 1`] = `
.bfYNTw {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
  bottom: -0.25em;
}

<sub
  className="sc-kpOJdX bfYNTw"
/>
`;

exports[`renders <Summary /> correctly 1`] = `
.chrtFK {
  display: list-item;
}

<summary
  className="sc-jDwBTQ chrtFK"
/>
`;

exports[`renders <Sup /> correctly 1`] = `
.hZcmXd {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
  top: -0.5em;
}

<sup
  className="sc-dxgOiQ hZcmXd"
/>
`;

exports[`renders <Svg /> correctly 1`] = `
.GnRoM:not(:root) {
  overflow: hidden;
}

<svg
  className="sc-hMqMXs GnRoM"
/>
`;

exports[`renders <Textarea /> correctly 1`] = `
.bDTEgL {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: auto;
}

<textarea
  className="sc-eHgmQL bDTEgL"
/>
`;

exports[`renders <Video /> correctly 1`] = `
.bkGlpl {
  display: inline-block;
}

<video
  className="sc-jKJlTe bkGlpl"
/>
`;
